man/demi_wait.md

```diff
@@ -8,14 +8,19 @@
 
 `demi_wait_any` - Waits for the first asynchronous I/O operation in a list to complete.
 
+`demi_timedwait_any` - Waits for an asynchronous I/O operation to complete or a timeout to expire.
+
 ## Synopsis
 
 ```c
 #include <demi/wait.h>
 #include <demi/types.h> /* For demi_qresult_t and demi_qtoken_t. */
 
 int demi_wait(demi_qresult_t *qr_out, demi_qtoken_t qt);
+int demi_timedwait(demi_qresult_t *qr_out, demi_qtoken_t qt, const struct timespec *abstime);
 int demi_wait_any(demi_qresult_t *qr_out, int *ready_offset, demi_qtoken_t qts[], int num_qts);
+int demi_timedwait_any(demi_qresult_t *qr_out, int *ready_offset, const demi_qtoken_t qts[], int num_qts,
+                                  const struct timespec *abstime);
```

Comment:
I still don't like "abstime".  At the very least, it should be "abs_time", although even then it isn't clear what "abs" is short for.

Reply:
Ratione for naming it as "abstime":
- To match POSIX documentation: https://pubs.opengroup.org/onlinepubs/9699919799/nframe.html
- To match Linux API: https://linux.die.net/man/3/pthread_cond_timedwait

Reply:
That opengroup link is to a generic landing page, so I'm not clear on what it is you're referencing.  And that "Linux API" link is actually describing a POSIX API, as mentioned in the "Prolog" section of that page.  At least that page makes it clear that "abstime" is short for "absolute time".

Should we violate our variable naming rules just to match some obscure POSIX API?  I'm not sure it matters much.    If you prefer it, that's fine.

Reply:
I like the idea of making the name unambiguous: e.g., absolute_time (it's not too long compared to ready_offset) but is very clear to the reader.